src/components/Top-Bar.vue

```diff
@@ -66,6 +66,18 @@ export default {
     if (this.intervalId) {
       clearInterval(this.intervalId)
     }
+  },
+  watch: {
+    isSidebarVisible: function (val) {
+      if (!val) {
+        /*
+        * Delay for screen readers Virtual buffers
+        */
+        setTimeout(() => {
+          this.$refs.menubutton.$el.focus()
+        }, 500)
```

Comment:
does it need to be a fixed delay value or can it be done with `this.$next()` ?

Reply:
500-100ms is a recommended value. setTimeout 0 or this.$next are too short, we have to wait for the Virtual Buffer of the screen reader. This can't be perceived via JS.

Reply:
since we are using this patterns in multiple places, I wonder if there is a way to make it more Vue-y.

Maybe implement something like `this.$next` but call it `this.$afterScreenReaderInit` or something so that it becomes self-explanatory ? The latter implementation can then contain whatever value we chose for the delay.

Reply:
`afterScreenReaderInit` sounds like an actual hook/lifecycle event. I would go with `delayForScreenreader` or the like. Emphasising that it is a form of guessing

Reply:
@marcus-herrmann your proposal sounds good

Reply:
@PVince81 So this would be eventually a Vue mixin which potentially every component can access, and it its core a wrapper around a setTimeout in order to not need to comment it?

Reply:
@marcus-herrmann yes, that would be a way.

One question that does come to mind is how can developers know when to use it and when not to. Does this need some experimentation with accessibility test systems or is there a common pattern ? If the latter, it could be included in the design guidelines in some form.

Reply:
@PVince81 It's always better to test/experiment with assistive technology, but I can devote a particular page in the design guidelines about screen readers and problems with web-apps (which are, predominantly, DOM-changing, asynchronously loading, stateful constructs, all of which makes them kind of hard to grasp for screen readers). Sending focus to elements in dynamic DOM parts of the application (like this, or after routing) would be just one part of the challenges, accessible notifications, live search and filtering would be others.